projects/core/src/cms/occ/occ-cms-component.adapter.ts

```diff
@@ -1,19 +1,20 @@
 import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
+import { catchError, pluck } from 'rxjs/operators';
+import { PageType } from '../../model/cms.model';
 import { CmsComponent, CmsComponentList } from '../../occ/occ-models/index';
 import { OccEndpointsService } from '../../occ/services/occ-endpoints.service';
 import { PageContext } from '../../routing/index';
-import { CmsComponentAdapter } from '../connectors/component/cms-component.adapter';
-import { IdList } from '../model/idList.model';
 import { ConverterService } from '../../util/converter.service';
+import { CmsComponentAdapter } from '../connectors/component/cms-component.adapter';
 import { CMS_COMPONENT_NORMALIZER } from '../connectors/component/converters';
-import { pluck } from 'rxjs/operators';
-import { PageType } from '../../model/cms.model';
+import { ComponentIds } from '../model/componentIds.model';
 
 @Injectable()
 export class OccCmsComponentAdapter implements CmsComponentAdapter {
   protected headers = new HttpHeaders().set('Content-Type', 'application/json');
+  private componentIdsFlag;
```

Comment:
better not create this class variable `componentIdsFlag`. You can add a flag in function `getComponentsRequestParams`. And `componentIdsFlag` is not a good name.